app/scripts/lib/xhr.js

```diff
@@ -67,6 +67,7 @@ define([
         // make sure to set the dataType for Firefox <21. See issue #1930
         dataType: 'json',
         type: options.type,
+        timeout: options.timeout || 0,
```

Comment:
Timeout set to 0 means no timeout, its the JQuery default when a timeout isnt set, so I thought we can use that here as well.